homeassistant/components/tradfri/fan.py

```diff
@@ -152,6 +152,8 @@ async def async_turn_on(
 
         if preset_mode:
             await self.async_set_preset_mode(preset_mode)
+        else
```

Comment:
I'd replace the whole if/else block with something like this:
```py
preset_mode = preset_mode or ATTR_AUTO
await self.async_set_preset_mode(preset_mode)
```

Reply:
Yes, thought about something similar too. I'll updated the PR.